build/common.ps1

```diff
@@ -268,22 +269,74 @@ Function Install-DotnetCLIToILMergePack {
     & $cli.DotNetExe --info
 }
 
+Function Get-LatestVisualStudioRoot() {
+    # First try to use vswhere to find the latest version of Visual Studio.
+    if (Test-Path $VsWhereExe) {
+        $installationPath = & $VsWhereExe -latest -prerelease -property installationPath
+        Verbose-Log "Found Visual Studio at '$installationPath' using vswhere"
+
+        return $installationPath
+    }
+
+    $cachePath = "${Env:ProgramData}\Microsoft\VisualStudio\Packages"
+
+    # Fall back to reading JSON directly (not recommended).
+    'Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\VisualStudio\Setup',
+    'Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\VisualStudio\Setup',
+    'Registry::HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\VisualStudio\Setup' | ForEach-Object {
+        if (Test-Path $_) {
+            $key = Get-ItemProperty $_
+
+            if ($key.CachePath) {
+                $cachePath = $key.CachePath
+                return
+            }
+        }
+    }
+
+    $instances = Get-ChildItem "$cachePath\_Instances" -Filter state.json -Recurse | Get-Content | ConvertFrom-Json
+    foreach ($instance in $instances) {
+        if (-not $maxInstance) {
+            $maxInstance = $instance
+        } elseif ([Version]$instance.installationVersion -gt [Version]$maxInstance.installationVersion) {
+            $maxInstance = $instance
+        } elseif (([Version]$instance.installationVersion -eq [Version]$maxInstance.installationVersion) -and ([DateTime]$instance.installDate -gt [DateTime]$maxInstance.installDate)) {
+            $maxInstance = $instance
+        }
+    }
+
+    if ($maxInstance) {
+        $installationPath = $maxInstance.installationPath
+        Verbose-Log "Found Visual Studio at '$installationPath' using machine configuration"
+
+        return $installationPath
+    }
+
+    Error-Log 'Cannot find an instance of Visual Studio 2017 or newer' -Fatal
+}
+
 Function Get-MSBuildRoot {
     param(
-        [ValidateSet(15)]
```

Comment:
This actually doesn't allow us to get the 15.x toolset and build, nor the 16x toolset and build. 
It's create some sort of an inconsistency with the configure.ps1 right now.